pkg/internal/database/database.go

```diff
@@ -20,19 +23,45 @@ type Database struct {
 }
 
 // NewDatabase will configure and return database based on provided config
-func NewDatabase(cfg *Config, logger *slog.Logger) (*Database, error) {
+func NewDatabase(cfg *defs.Database, logger *slog.Logger) (*Database, error) {
+	defaultConfig := defs.DefaultDBConfig()
+	if cfg == nil {
+		return newDatabaseInternal(defaultConfig, logger)
+	}
+
+	mergedCfg := mergeConfig(defaultConfig, cfg)
+	return newDatabaseInternal(mergedCfg, logger)
+}
+
+// newDatabaseInternal configures database with merged default and provided config
+func newDatabaseInternal(cfg *defs.Database, logger *slog.Logger) (*Database, error) {
 	var database *gorm.DB
 
 	if logger == nil {
 		logger = slog.Default()
 	}
```

Comment:
```suggestion
	logger = logging.Child(logger, "database")
```
This also makes `slog.Default` if provided logger is nil.